pyquil/quilbase.py

```diff
@@ -135,6 +135,23 @@ def controlled(self, control_qubit):
 
         return self
 
+    def forked(self, fork_qubit, alt_params):
+        """
+        Add the FORKED modifier to the gate with the given fork qubit and given additional parameters.
```

Comment:
I think more thorough documentation would be much appreciated by those not in-the-know, particularly an example.

Reply:
I put it in the `basics.rst` doc. I'm not opposed to basically pasting that here as well, but I took stylistic guidance from the surrounding code.

Reply:
That's fine. I didn't see that when I made the comment.